pkg/internal/token/azuredevelopercli.go

```diff
@@ -0,0 +1,81 @@
+package token
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+type AzureDeveloperCLIToken struct {
+	resourceID string
+	tenantID   string
+	timeout    time.Duration
+}
+
+// newAzureDeveloperCLIToken returns a TokenProvider that will fetch a token for the user currently logged into the Azure CLI.
+// Required arguments include an oAuthConfiguration object and the resourceID (which is used as the scope)
+func newAzureDeveloperCLIToken(resourceID string, tenantID string, timeout time.Duration) (TokenProvider, error) {
+	if resourceID == "" {
+		return nil, errors.New("resourceID cannot be empty")
+	}
+
+	if timeout <= 0 {
+		timeout = defaultTimeout
+	}
+
+	return &AzureDeveloperCLIToken{
+		resourceID: resourceID,
+		tenantID:   tenantID,
+		timeout:    timeout,
+	}, nil
+}
+
+// Token fetches an azcore.AccessToken from the Azure CLI SDK and converts it to an adal.Token for use with kubelogin.
+func (p *AzureDeveloperCLIToken) Token(ctx context.Context) (adal.Token, error) {
+	emptyToken := adal.Token{}
+
+	// Request a new Azure CLI token provider
+	cred, err := azidentity.NewAzureDeveloperCLICredential(&azidentity.AzureDeveloperCLICredentialOptions{
```

Comment:
you probably copied the pattern from azurecli implementation. the credential should be created in newFunction like https://github.com/Azure/kubelogin/blob/main/pkg/internal/token/federatedIdentity.go#L59. it didn't have any material impact until recently we enabled library usage.

Reply:
The internal credential is being created via the updated `azidentity` package and `NewAzureDeveloperCLICredential` new function.  Let me know what you are expecting to be different here.

Reply:
since in library use client may keep the token provider longer and call `Token()` as many times as needed, it's better to save the `cred` in `AzureDeveloperCLIToken` struct

Reply:
your PR is not adding this login to library use, which is fine. but i would still like this credential  to be created in new function. :)